cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java

```diff
@@ -1158,15 +1154,16 @@ public ArtifactData deserialize(JsonElement json,
       // For Backward Compatibility
       if (jsonObj.has("locationURI")) {
         try {
-          URI locationURI = new URI(jsonObj.get("locationURI").getAsString());
+          URI locationURI = context.deserialize(jsonObj.get("locationURI"), URI.class);
           locationPath = locationURI.getPath();
         } catch (Exception e) {
```

Comment:
now that you're using context.deserialize which only throws JsonParseException, you don't need a catch block.